libs/stanag4609/src/main/java/org/codice/alliance/libs/stanag4609/MetadataPacket.java

```diff
@@ -16,19 +16,33 @@
 import org.codice.ddf.libs.klv.KlvDecoder;
 import org.jcodec.containers.mps.MPSDemuxer.PESPacket;
 
-class AsynchronousMetadataPacket extends AbstractMetadataPacket {
-    private static final int ASYNCHRONOUS_PES_PACKET_HEADER_LENGTH = 9;
+class MetadataPacket extends AbstractMetadataPacket {
+    private static final int PES_PACKET_HEADER_LENGTH = 9;
 
-    AsynchronousMetadataPacket(final byte[] pesPacketBytes, final PESPacket pesHeader,
+    /**
+     * Location of the "PES header length" field which contains the number of
+     * extra bytes contained in the header.
+     */
+    private static final int PES_HEADER_LENGTH_INDEX = 8;
+
+    MetadataPacket(final byte[] pesPacketBytes, final PESPacket pesHeader,
             final KlvDecoder decoder) {
         super(pesPacketBytes, pesHeader, decoder);
     }
 
     @Override
     protected byte[] getKLVBytes() {
-        // For asynchronous metadata streams, the header is supposed to be 9 bytes long. The header's
-        // length field gives the number of bytes in the packet following it, so we need to skip
+        // For metadata streams, the base header is supposed to be 9 bytes long. The header's
+        // length field gives the number of extra bytes in the packet following it, so we need to skip
         // the 3 header bytes after the length field to get the true length of the payload.
-        return getPESPacketPayload(pesHeader.length - 3, ASYNCHRONOUS_PES_PACKET_HEADER_LENGTH);
+
+        if (this.pesPacketBytes.length < PES_PACKET_HEADER_LENGTH) {
+            return null;
+        }
+
+        int additionalHeaderBytes = this.pesPacketBytes[PES_HEADER_LENGTH_INDEX];
+
+        return getPESPacketPayload(pesHeader.length - 3,
```

Comment:
We can no longer assume there are only 3 header bytes following the payload length byte. We need to subtract `additionalHeaderBytes` as well.